freebites/product
# 50 delete during post creation (merge with 47)

# Description

We revamped the modal popup in the post creation window, and added the functionality to the delete button in this modal. We also updated the structure of the modals throughout the app to be more modular and efficient.

## Issues






Resolve #50 

## Screenshots


<img src="https://github.com/user-attachments/assets/3668ea3f-a22d-423e-8463-adaa7dc8863b" width=200 />
<img src="https://github.com/user-attachments/assets/07e27298-af1f-4cd7-af40-a5d8df9404cf" width=200 />

## Test
- log in
- click on the create new post button on the home screen
- take a picture to use for the post
- Continue into the post creation page, however many screens you want to move through
- click the three dots in the top right of the screen, the modal should appear
- press the background or cancel button to close the modal
- press the dots to reopen it
- press the delete button and you should be returned to the home screen with no new post created.

## Possible Downsides

There may still be some references in the code still to the previous implementation of this modal that we missed or failed to remove. Creating new modals will require the use of the FreeBitesModal component from now on as opposed to the react native Modal component. 

## Additional Documentations
